### PR TITLE
[Generate] Greedy Search, fix output scores from logits to scores

### LINKED ISSUE #17424: Inconsistent behavior in generate when output_scores=True
### System Info

```shell
master branch
```

### Who can help?

@patrickvonplaten, @Narsil, @gante

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In `generate` when `output_scores=True`, the behavior is inconsistent. In `greedy_search` mode, the scores are raw logits https://github.com/huggingface/transformers/blob/main/src/transformers/generation_utils.py#L1694, but in `sample` mode (and various beam search modes), the scores are processed logits https://github.com/huggingface/transformers/blob/main/src/transformers/generation_utils.py#L1953.

### Expected behavior

```shell
In `generate` when `output_scores=True`, the returned scores should be consistent. It could either be raw logits or the processed logits. While for my usecase, I only need raw logits. There might be some usecase which requires the processed logits. So there're multiple solutions:

1. Return raw logits when `output_scores=True`
2. Return processed logits when `output_scores=True`
3. Return processed logits when `output_scores=True`, and raw logits when `output_raw_scores=True`
```

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

🚨🚨  **This PR can lead to silently changing values for users of `greedy_search` and `output_scores=True`. Please read the issue below** 🚨 🚨 

Fixes https://github.com/huggingface/transformers/issues/17424

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Good for me, so that the output is consistently scores.

**Review by patil-suraj — APPROVED** [review]
LGTM!

>🚨🚨 This PR can lead to silently changing values for users of greedy_search and output_scores=True. Please read the issue below 🚨 🚨

went through the issue, and it's fine for me since this will actually return the correct scores.

**Comment by xkianteb** [comment]
@patil-suraj Will there be any support for raw logits instead logits that are processed? (see #17521 )
